module_3/gift_certificates_system/model/src/main/java/com/epam/esm/entity/Tag.java

```diff
@@ -0,0 +1,51 @@
+package com.epam.esm.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import java.io.Serializable;
+
+/**
+ * Tag entity.
+ */
+@RequiredArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@Entity
+@ToString
+public class Tag implements Serializable {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false, unique = true)
+    private String name;
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+
+        if (o == null || getClass() != o.getClass()) return false;
+
+        Tag tag = (Tag) o;
+
+        return new EqualsBuilder().append(getName(), tag.getName()).isEquals();
+    }
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder(17, 37).append(getName()).toHashCode();
+    }
```

Comment:
Why didn't you define this via lombok?

Reply:
Using `@EqualsAndHashCode` for JPA entities is not recommended. It can cause severe performance and memory consumption issues.